mount/mount_unix_test.go

```diff
@@ -230,3 +234,43 @@ func TestRecursiveUnmountTooGreedy(t *testing.T) {
 		t.Fatal("expected dir-other to be mounted, but it's not")
 	}
 }
+
+func TestRecursiveUnmount_SubMountFailsToUnmount(t *testing.T) {
+	if os.Getuid() != 0 {
+		t.Skip("root required")
+	}
+
+	tmp := t.TempDir()
+
+	dir := "sub1/sub2/sub3"
+	err := os.MkdirAll(path.Join(tmp, dir), 0o700)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	grandchild := path.Join(tmp, dir)
+	child := filepath.Dir(grandchild)
+	parent := filepath.Dir(child)
+
+	// order is necessary to create shadow mounts.
+	toMount := []string{grandchild, child, parent}
+	for _, dir := range toMount {
+		dir := dir
+		if err := Mount("tmpfs", dir, "tmpfs", ""); err != nil {
+			t.Fatal(err)
+		}
+		//nolint:errcheck
+		defer Unmount(dir)
+	}
+
+	// unmount shadowed mounts
+	shadowedMounts := []string{child, grandchild}
+	for _, shadowedMount := range shadowedMounts {
+		// Unmount dir, make sure dir-other is still mounted.
```

Comment:
This comment no longer reflects reality -- I don't see "dir" and "dir-other" in here.